YIF.Core.Service/Concrete/Services/IoEAdminService.cs

```diff
@@ -58,25 +58,38 @@ ResourceManager resourceManager
             _configuration = configuration;
         }
 
-        public async Task<ResponseApiModel<DescriptionResponseApiModel>> AddSpecialtyToIoe(
-            SpecialtyToInstitutionOfEducationPostApiModel specialtyToIoE)
+        public async Task<ResponseApiModel<DescriptionResponseApiModel>> AddRangeSpecialtiesToIoE(
+            IEnumerable<SpecialtyToInstitutionOfEducationPostApiModel> specialtyToIoE)
         {
             var result = new ResponseApiModel<DescriptionResponseApiModel>();
 
-            var specialty = await _specialtyRepository.ContainsById(specialtyToIoE.SpecialtyId);
-            var institutionOfEducation = await _ioERepository.ContainsById(specialtyToIoE.InstitutionOfEducationId);
-
-            var specialtyToInstitutionOfEducationDTO = _mapper.Map<SpecialtyToInstitutionOfEducationDTO>(specialtyToIoE);
-            var specialtyToInstitutionOfEducation = _mapper.Map<SpecialtyToInstitutionOfEducation>(specialtyToInstitutionOfEducationDTO);
-
-            if (institutionOfEducation == false)
-                throw new BadRequestException(_resourceManager.GetString("InstitutionOfEducationNotFound"));
-
-            if (specialty == false)
-                throw new BadRequestException(_resourceManager.GetString("SpecialtyNotFound"));
-
-            await _specialtyToIoERepository.AddSpecialty(specialtyToInstitutionOfEducation);
-            return result.Set(new DescriptionResponseApiModel("Specialty was successfully added to the Institution of Education"), true);
+            foreach (var item in specialtyToIoE)
+            {
+                var specialtyToInstitutionOfEducationDTO = _mapper.Map<SpecialtyToInstitutionOfEducationDTO>(item);
+                var specialtyToInstitutionOfEducation = _mapper.Map<SpecialtyToInstitutionOfEducation>(specialtyToInstitutionOfEducationDTO);
+                
+                SpecialtyToInstitutionOfEducation specialtyToInstitutionOf = new SpecialtyToInstitutionOfEducation
+                {
+                    SpecialtyId = specialtyToInstitutionOfEducation.SpecialtyId,
+                    InstitutionOfEducationId = specialtyToInstitutionOfEducation.InstitutionOfEducationId,
+                    IsDeleted = false
+                };
+
+                var id = await _specialtyToIoERepository.AddSpecialty(specialtyToInstitutionOf);
```

Comment:
non-informative variable name, maybe specialtyId will be better

Reply:
done